crates/relayer-cli/src/cli_utils.rs

```diff
@@ -102,3 +102,88 @@ pub fn spawn_chain_counterparty<Chain: ChainHandle>(
         channel_connection_client,
     ))
 }
+
+/// Check that the relayer can send on the given channel and ensure that channels and chain identifiers match.
+/// To do this, fetch from the source chain the channel end, then the associated connection
+/// end, and then the underlying client state; finally, check that this client is verifying
+/// headers for the destination chain.
```

Comment:
I think these comments would be more helpful as annotations in the body of the function.